service/src/main/java/greencity/service/HabitStatusServiceImpl.java

```diff
@@ -120,6 +125,25 @@ private void updateHabitStatus(HabitStatus habitStatus, LocalDate todayDate) {
         }
     }
 
+    /**
+     * Method updates {@link HabitAssign} if it's completed.
+     *
+     * @param habitStatus {@link HabitStatus} instance.
+     */
+    private void checkAssignCompleted(HabitStatus habitStatus) {
+        int workingDays = habitStatus.getWorkingDays();
+        int habitDuration = habitStatus.getHabitAssign().getDuration();
+        if (workingDays == habitDuration) {
+            if (habitStatus.getHabitAssign().getAcquired().equals(true)) {
```

Comment:
Boolean.TRUE.equals(...) this will help to avoid nullpointer

Reply:
Changes done.